webapi/src/main/scala/org/knora/webapi/responders/admin/ListsResponderADM.scala

```diff
@@ -1830,6 +1832,46 @@ class ListsResponderADM(responderData: ResponderData) extends Responder(responde
 
     } yield CanDeleteListResponseADM(iri, canDelete)
 
+  /**
+   * Deletes all comments from child list node.
+   */
+  private def deleteListChildNodeCommentsADM(
+    iri: IRI,
+    featureFactoryConfig: FeatureFactoryConfig,
+    requestingUser: UserADM
+  ): Future[ListChildNodeCommentsDeleteResponseADM] =
+    for {
+      node <- listNodeGetADM(
+                nodeIri = iri,
+                shallow = false,
+                featureFactoryConfig = featureFactoryConfig,
+                requestingUser = KnoraSystemInstances.Users.SystemUser
+              )
+
+      isRootNode = node match {
+                     case Some(_: ListRootNodeADM)  => true
+                     case Some(_: ListChildNodeADM) => false
+                     case _                         => false
```

Comment:
this case should probably never happen, right? then throwing an error might be more in line with how there cases are handled in the rest of the code base

Reply:
Indeed this line seems to be redundant, however I added an exception throw.